ale_linters/elixir/mix.vim

```diff
@@ -0,0 +1,52 @@
+" Author: evnu - https://github.com/evnu
+" Author: colbydehart - https://github.com/colbydehart
+" Description: Mix compile checking for Elixir files
+
+function! ale_linters#elixir#mix#Handle(buffer, lines) abort
+    " Matches patterns like the following:
+    "
+    " Error format
+    " ** (CompileError) apps/sim/lib/sim/server.ex:87: undefined function update_in/4
+    "
+    " TODO: Warning format
+    " warning: variable "foobar" does not exist and is being expanded to "foobar()", please use parentheses to remove the ambiguity or change the variable name
+
+    let l:pattern = '\v\(([^\)]+Error)\) ([^:]+):([^:]+): (.+)$'
+    let l:output = []
+
+    for l:match in ale#util#GetMatches(a:lines, l:pattern)
+        let l:type = 'E'
+        let l:text = l:match[4]
+
+        call add(l:output, {
+        \   'bufnr': a:buffer,
+        \   'lnum': l:match[3] + 0,
+        \   'col': 0,
+        \   'type': l:type,
+        \   'text': l:text,
+        \})
+    endfor
+
+    return l:output
+endfunction
+
+function! ale_linters#elixir#mix#GetCommand(buffer) abort
+    let l:project_root = fnamemodify(ale#path#FindNearestFile(a:buffer, 'mix.exs'), ':h')
+
+    let l:temp_dir = ale#engine#CreateDirectory(a:buffer)
+
+    let l:mix_build_path = has('win32')
+    \   ? 'set MIX_BUILD_PATH=' . ale#Escape(l:temp_dir) . ' && '
+    \   : 'MIX_BUILD_PATH=' . ale#Escape(l:temp_dir) . ' '
+
+    return  ale#path#CdString(l:project_root)
+          \ . l:mix_build_path
+          \ . ' mix compile %s'
+endfunction
```

Comment:
Add some tests for the output of this function here. See similar tests in `test/command_callback`.

Reply:
sorry, been traveling. I couldn't find an example of how to test a command function where you make a temporary directory. Could you maybe point me in the right direction of how to do that?

Reply:
I found one for javac and did what they did for testing here https://github.com/w0rp/ale/pull/1618/files#diff-9a91b3bcd6181a1601673b8b6c8999a6